### PR TITLE
Improve observability by integrating sentry, datadog, and adding extended logging

### PR DESCRIPTION
**Pull request by bdach**
- [x] Depends on https://github.com/ppy/osu-server-beatmap-submission/pull/14

Not much interaction with datadog here other than incrementing a counter for uploads similar to old BSS. If you have ideas what would be nice to see there I'm all ears.

### CODE REVIEW

File: osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs [hunk 1]
```diff
@@ -60,13 +62,15 @@ public async Task StoreBeatmapSetAsync(uint beatmapSetId, byte[] beatmapPackage,
 
             await Task.WhenAll([
                 uploadBeatmapPackage(beatmapSetId, beatmapPackage, stream),
-                ..uploadAllVersionedFiles(allFiles),
-                ..uploadAllBeatmapFiles(beatmapFiles)
+                ..uploadAllVersionedFiles(beatmapSetId, allFiles),
```
Comment (peppy) [inline osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs:65]: I know I already let this syntax in (by oversight) but I am not sure I can get used to this. It's quite non-obvious what this is doing.

```diff
diff --git a/osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs b/osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs
index f78441b..cbce08e 100644
--- a/osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs
+++ b/osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs
@@ -60,11 +60,10 @@ namespace osu.Server.BeatmapSubmission.Services
                     beatmapFiles.Add((beatmapId, contents));
             }
 
-            await Task.WhenAll([
-                uploadBeatmapPackage(beatmapSetId, beatmapPackage, stream),
-                ..uploadAllVersionedFiles(beatmapSetId, allFiles),
-                ..uploadAllBeatmapFiles(beatmapSetId, beatmapFiles)
-            ]);
+            await uploadBeatmapPackage(beatmapSetId, beatmapPackage, stream);
+            await Task.WhenAll(uploadAllVersionedFiles(beatmapSetId, allFiles));
+            await Task.WhenAll(uploadAllBeatmapFiles(beatmapSetId, beatmapFiles));
+
             logger.LogInformation("All file uploads for beatmapset {beatmapSetId} concluded successfully.", beatmapSetId);
         }
 
@@ -165,7 +164,8 @@ namespace osu.Server.BeatmapSubmission.Services
             {
                 foreach (var file in files)
                 {
-                    using var response = await client.GetObjectAsync(AppSettings.S3CentralBucketName, getPathToVersionedFile(BitConverter.ToString(file.File.sha2_hash).Replace("-", string.Empty).ToLowerInvariant()));
+                    using var response = await client.GetObjectAsync(AppSettings.S3CentralBucketName,
+                        getPathToVersionedFile(BitConverter.ToString(file.File.sha2_hash).Replace("-", string.Empty).ToLowerInvariant()));
                     zipWriter.Write(file.VersionFile.filename, response.ResponseStream);
                 }
             }

```

is probably what i'd stick with.

**Review by peppy — COMMENTED** [review]
Looks pretty good apart from kinda unrelated syntax thing.

File: osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs (hunk 1 above)
Comment (peppy) [reply osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs]: I'd still just use multiple awaits for better readability, but I'm not going to get hung up on it.

**Review by peppy — APPROVED** [review]

File: osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs (hunk 1 above)
Comment (bdach) [reply osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs]: I sort of explicitly didn't want to do multiple awaits, as that reduces concurrency :shrug:

Comment (peppy) [reply osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs]: Fair, I wasn't considering that part.

**Pull request merged**